app/src/main/java/com/github/se/icebreakrr/model/location/GeoFirestoreRepository.kt

```diff
@@ -0,0 +1,34 @@
+package com.github.se.icebreakrr.model.location
+
+import android.util.Log
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.GeoPoint
+import org.imperiumlabs.geofirestore.GeoFirestore
+import org.imperiumlabs.geofirestore.extension.setLocation
+
+class GeoFirestoreRepository(
+    private val geoFirestore: GeoFirestore =
+        GeoFirestore(FirebaseFirestore.getInstance().collection("profiles")),
+    private val auth: FirebaseAuth = FirebaseAuth.getInstance(),
+    private val onSetLocationComplete: (Exception?) -> Unit = {}
+) {
+
+  /**
+   * Updates the user's position in Firestore using GeoFirestore.
+   *
+   * This function retrieves the current user ID from the authentication instance and, if
+   * authenticated, sets the user's position in Firestore to the specified `newLocation`. If the
+   * update fails, an error message is logged.
+   *
+   * @param newLocation The new geographic location of the user, represented as a `GeoPoint`.
+   */
+  fun setUserPosition(newLocation: GeoPoint) {
+    val userId = auth.currentUser?.uid
+    if (userId == null) {
+      Log.w("GeoFirestoreRepository", "User is not authenticated. Cannot set location.")
+      return
+    }
+    geoFirestore.setLocation(userId, newLocation, onSetLocationComplete)
```

Comment:
What happens here when the user refused to give his location at the beginning of the app ? Because the user could be login but refuse to give his location. I am maybe wrong but I don't see it being hadled anywhere in the new code

Reply:
This function does not require location permissions; it will simply perform an operation via geoFirestore to store the new position. It will be called in the onLocationUpdate callback, which is passed to the startLocationUpdates function of the LocationService. The startLocationUpdates function is invoked from LocationViewModel, which ensures that location permissions are obtained.